lib/virt_autotest/virtual_network_utils.pm

```diff
@@ -331,4 +331,17 @@ sub get_active_pool_and_available_space {
     return ($active_pool, $available_size);
 }
 
+sub clean_all_virt_networks {
+    my $_virt_networks = script_output("virsh net-list --name --all", 30, type_command => 0, proceed_on_failure => 0);
+
+    foreach my $vnet (split(/\n+/, $_virt_networks)) {
+        my $_br = script_output(q@virsh net-dumpxml @ . $vnet . q@|grep -o "bridge name=[^\s]*" | sed  's#bridge name=##'@, type_command => 0, proceed_on_failure => 0);
+        script_run("virsh net-destroy $vnet");
+        assert_script_run("virsh net-undefine $vnet");
+        assert_script_run("if ip a|grep $_br;then ip link del $_br;fi");
```

Comment:
I think you can
```
ip addr flush
ip link down
ip link del
```
The sequence looks to me more elegant.

Reply:
Maybe more elegant, but more unnecessary? The last command directly works...